### PR TITLE
Documentation for the testing environment

### PR DESCRIPTION
**Pull request by vilda**

### CODE REVIEW

**Comment by vilda** [comment]
Can you check this @StephenHynes7 also?

File: test/README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
+Logentries Agent Tests
+======================
+
+This directory contains a simple testing framework together with a set of tests. It provides a partially virutalized environment for speed and simplicity.
+
+All features must be covered.
+
+Tests are located in `tests.d` directory. Particular feature must be in a separate file. If there are too many test cases, it's advisable to split them into multiple files.
```
Comment (StephenHynes7) [inline test/README.md]: 'Specific features are split into separate files'

Something like that?

File: test/README.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+Or run selected test only by specifying file name:
+
+	./tests.sh tests.d/formatters.sh
+
+Or, if you are interested in full output, run the test directly:
+
+	tests.d/formatters.sh
+
+The `testd.sh` script extracts expected output and compares it against actual output. Their difference are displayed then; unless they are the same in which case `SUCCESS` is displayed.
```
Comment (StephenHynes7) [inline test/README.md]: 'The difference is then displayed'

File: test/README.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+Typically test starts defining a scenario:
+
+	Scenario 'Default formatters use'
+
+Then a set of test cases:
+
+	Testcase 'Default formatter for HTTP PUT is plain'
+
+Each scenario clears configuration files, but leaves others untouched. Test cases does not modify the environment.
```
Comment (StephenHynes7) [inline test/README.md]: 'Test cases do not modify the environment'

File: test/README.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+Expected output is specified as bash comment `#` followed by `e` for standard error or `o` for standard output. ISO dates must be replaced with `ISODATETIME` keyword. Since home directory is generated, replace any occurrence with `$TMP`.
```
Comment (StephenHynes7) [inline test/README.md]: Expected output is specified as a bash comment

**Comment by StephenHynes7** [comment]
Just a few small changes, otherwise it looks fine.

**Comment by vilda** [comment]
Thanks @StephenHynes7, I fixed it.

**Pull request merged**